aws/resource_aws_macie2_custom_data_identifier_test.go

```diff
@@ -0,0 +1,367 @@
+package aws
+
+import (
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/macie2"
+	"github.com/hashicorp/aws-sdk-go-base/tfawserr"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/acctest"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+	"github.com/terraform-providers/terraform-provider-aws/aws/internal/naming"
+)
+
+func testAccAwsMacie2CustomDataIdentifier_basic(t *testing.T) {
+	var macie2Output macie2.GetCustomDataIdentifierOutput
+	resourceName := "aws_macie2_custom_data_identifier.test"
+	regex := "[0-9]{3}-[0-9]{2}-[0-9]{4}"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckAwsMacie2CustomDataIdentifierDestroy,
+		ErrorCheck:        testAccErrorCheck(t, macie2.EndpointsID),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccAwsMacieCustomDataIdentifierconfigNameGenerated(regex),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAwsMacie2CustomDataIdentifierExists(resourceName, &macie2Output),
```

Comment:
The test should validate the value of the `arn` field. We have a number of functions to do this, matching something like `testAccCheckResourceAttrRegionalARN...` or `testAccMatchResourceAttrRegionalARN...`